lib/grandpa/commits_tracker.go

```diff
@@ -14,6 +15,7 @@ import (
 // its maximum capacity is reached.
 // It is NOT THREAD SAFE to use.
 type commitsTracker struct {
+	sync.Mutex
```

Comment:
please don't embed the mutex to avoid exposing `.Lock()` and `.Unlock()` methods (even though it's not exported for now, it might be in the future). It also makes asserting the entire object cumbersome in tests as you've seen.

Reply:
I don't mind embedding the mutex in private types.  In the event that an instance of `commitsTracker` needs to be public in the future, we can revise the embedding then.

Reply:
Sure, but there is really no need to embed it